File: modules/pico-module-maven-plugin/src/test/java/com/picocontainer/testmodules/moduleOne/Composition.java
/**
 * 
 */
package com.picocontainer.testmodules.moduleOne;

import com.picocontainer.modules.Publisher;
import com.picocontainer.modules.deployer.AbstractPicoComposer;

import com.picocontainer.MutablePicoContainer;

public class Composition extends AbstractPicoComposer {

	public Composition() {
		super();
	}

	@Override
	protected MutablePicoContainer populateChildContainer(
			final MutablePicoContainer childContainer,
			final MutablePicoContainer parent, final Object assemblyScope) {

		childContainer.addComponent("moduleOneTest")
			.addComponent(ServiceOne.class.getName(), DefaultServiceOne.class);

		new Publisher(childContainer, parent)
			.publish(ServiceOne.class.getName());
			
		
//		final ComponentAdapter<ServiceOne> ca = childContainer
//				.getComponentAdapter(ServiceOne.class, (NameBinding) null);
//
//		parent.addAdapter(new Publishing<ServiceOne>(childContainer, ca));

		return childContainer;
	}

}

File: sandbox/modules/pico-module-maven-plugin/src/test/modules/com.picocontainer.testmodules.moduleOne/META-INF/composition.js
importPackage(Packages.com.picocontainer)
importPackage(Packages.com.picocontainer.modules)

var pico = parent.makeChildContainer()
		.addComponent("moduleOneTest")
		.addComponent("ServiceOne",com.picocontainer.testmodules.moduleOne.DefaultServiceOne);
	
	new Publisher(pico, parent)
		.publish("ServiceOne");

	
	//var ca = pico.getComponentAdapter("ServiceTwo");
	
	//Publish service in parent container
	//parent.addAdapter(new Publishing(pico, ca));
